src/Cake.GitHub/Issues/GitHubIssueUpdater.cs

```diff
@@ -0,0 +1,131 @@
+            {
+                _cakeLog.Verbose($"{displayName} is not yet assigned to a Milestone, setting to Milestone {milestone.Number} '{milestone.Title}'");
+                await _githubClient.Issue.Update(owner, repository, number, new IssueUpdate() { Milestone = milestone.Number });
+            }
+            else
+            {
+                if (issue.Milestone.Number == milestone.Number)
+                {
+                    _cakeLog.Verbose($"{displayName} is already assigned to Milestone {milestone.Number} '{milestone.Title}'");
+                    return;
+                }
+                else if (settings.Overwrite)
+                {
+                    _cakeLog.Verbose($"Reassigning {displayName} Milestone {milestone.Number} '{milestone.Title}' because the 'Overwrite' setting was set to true.");
+                    await _githubClient.Issue.Update(owner, repository, number, new IssueUpdate() { Milestone = milestone.Number });
+                }
+                else
+                {
+                    throw new MilestoneAlreadySetException($"{displayName} is already assigned to Milestone {issue.Milestone.Number} '{issue.Milestone.Title}'");
+                }
+            }
+        }
+
+        private void LogSettings(string owner, string repository, int number, string milestoneTitle, GitHubSetMilestoneSettings settings)
+        {
+            const int padding = -29;
+
+            _cakeLog.Debug("Setting GitHub Milestone with the following settings:");
+            _cakeLog.Debug($"\t{"Owner",padding}: '{owner}'");
+            _cakeLog.Debug($"\t{"Repository",padding}: '{repository}'");
+            _cakeLog.Debug($"\t{"Issue or Pull Requets Number",padding}: '{number}'");
+            _cakeLog.Debug($"\t{"MilestoneTitle",padding}: '{milestoneTitle}'");
+            _cakeLog.Debug($"\t{nameof(settings.Overwrite),padding}: '{settings.Overwrite}'");
+            _cakeLog.Debug($"\t{nameof(settings.CreateMilestone),padding}: '{settings.CreateMilestone}'");
+        }
+
+        private async Task<Milestone> GetOrCreateMilestoneAsync(string owner, string repository, string milestoneTitle, GitHubSetMilestoneSettings settings)
+        {
+            _cakeLog.Verbose($"Looking up Milestone with title '{milestoneTitle}'");
+
+            // Get all milestones and filter client-side since we only know the name and not the milestone number
+            var milestones = await _githubClient.Issue.Milestone.GetAllForRepository(owner, repository, new MilestoneRequest() { State = ItemStateFilter.All });
+            var milestone = milestones.SingleOrDefault(x => StringComparer.Ordinal.Equals(x.Title, milestoneTitle));
+
+            if (milestone == null)
+            {
+                if (settings.CreateMilestone)
+                {
+                    _cakeLog.Verbose($"No Milestone titled '{milestoneTitle}' was not found in repository {owner}/{repository}. Creating new milestone.");
```

Comment:
log message seems incorrect, please either remove the `No` or the `not`

Reply:
Fixed with d947e41be5e65dc7a4304371bb6f690e03f21028